pkg/crd/crd.go

```diff
@@ -427,10 +432,19 @@ func (c *Client) RemoveAssignedIdentity(assignedIdentity *aadpodid.AzureAssigned
 		c.reporter.Report(
 			metrics.AssignedIdentityDeletionCountM.M(1),
 			metrics.AssignedIdentityDeletionDurationM.M(metrics.SinceInSeconds(begin)))
-
 	}()
 
-	err = c.rest.Delete().Namespace(assignedIdentity.Namespace).Resource("azureassignedidentities").Name(assignedIdentity.Name).Do().Error()
+	var res aadpodv1.AzureAssignedIdentity
+	err = c.rest.Delete().Namespace(assignedIdentity.Namespace).Resource(aadpodid.AzureAssignedIDResource).Name(assignedIdentity.Name).Do().Into(&res)
+	if err != nil && apierrors.IsNotFound(err) {
+		return nil
+	}
+	if err == nil && HasFinalizer(res.ObjectMeta) {
```

Comment:
why do we have to check if there is a finalizer. Are there cases where it could not be there. Is that for backward compat ? If so could you please add a comment.

Reply:
There are 2 reasons it's possible -
1. It's an old azure assigned identity
2. The finalizer was manually removed

This check is just to see if we need to remove the finalizer if exists, so we can unblock garbage collection of deleted azureassignedidentity. Will add this as a comment.